android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/client/HealthVaultRestClient.java

```diff
@@ -0,0 +1,136 @@
+ */
+
+package com.microsoft.hsg.android.simplexml.client;
+
+import java.io.IOException;
+import java.net.URL;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Observable;
+import java.util.concurrent.TimeUnit;
+
+import org.joda.time.DateTime;
+import org.joda.time.Minutes;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.util.Base64;
+
+import com.microsoft.hsg.Authenticator;
+import com.microsoft.hsg.Connection;
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.Transport;
+import com.microsoft.hsg.URLConnectionTransport;
+import com.microsoft.hsg.android.simplexml.HealthVaultSettings;
+import com.microsoft.hsg.android.simplexml.HealthVaultFileSettings;
+import com.google.common.reflect.TypeToken;
+import com.microsoft.rest.RestClient;
+import com.microsoft.rest.ServiceCallback;
+import com.microsoft.rest.ServiceClient;
+import com.microsoft.rest.ServiceFuture;
+import com.microsoft.rest.ServiceResponse;
+import com.microsoft.rest.Validator;
+
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+
+import healthvault.client.models.ActionPlan;
+import healthvault.client.models.ActionPlansResponseActionPlanInstance;
+import healthvault.client.models.Objective;
+import okhttp3.ConnectionPool;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.GET;
+import retrofit2.http.HTTP;
+import retrofit2.http.Headers;
+import retrofit2.http.PATCH;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+
+public class HealthVaultRestClient implements IHealthVaultRestClient {
+
+	/** The instance. */
+	private static MicrosoftHealthVaultRestApiImpl instance;
+	private DateTime lastRefreshedSessionCredential;
+	private final int SessionCredentialCallThresholdMinutes = 5;
+
+	public MicrosoftHealthVaultRestApiImpl getInstance(HealthVaultSettings settings, Connection connection) {
+		if (instance == null) {
+			final String restURL = settings.getRestUrl();
+			final Connection hvConnection = connection;
+			tokenRefreshCheck(hvConnection);
+			instance = new MicrosoftHealthVaultRestApiImpl(restURL, getOkHttp(hvConnection), getRetrofit(restURL).newBuilder());
+		}
+
+		return instance;
+	}
+
+	private Retrofit getRetrofit(String url){
+		Retrofit retrofit = new Retrofit.Builder()
+				.baseUrl(url)
+				.addConverterFactory(GsonConverterFactory.create())
```

Comment:
Is jackson json converting as well? If so why do we have two different json converters?